src/internal/TouchRipple.js

```diff
@@ -3,16 +3,11 @@ import ReactDOM from 'react-dom';
 import ReactTransitionGroup from 'react-addons-transition-group';
 import Dom from '../utils/dom';
 import CircleRipple from './CircleRipple';
-import update from 'react-addons-update';
-
-function push(array, obj) {
-  const newObj = Array.isArray(obj) ? obj : [obj];
-  return update(array, {$push: newObj});
-}
 
 function shift(array) {
-  // Remove the first element in the array using React immutability helpers
-  return update(array, {$splice: [[0, 1]]});
+  const [...newArray] = array;
+  newArray.splice(0, 1);
+  return newArray;
```

Comment:
@newoga - please leave the first part of the comment (if you weren't already going to), as that syntax could be mistaken for an error at first glance (and would be in older browsers...).